codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoder.java

```diff
@@ -314,11 +327,13 @@ public void onPriorityRead(ChannelHandlerContext ctx, int streamId, int streamDe
             Http2Stream stream = connection.stream(streamId);
 
             try {
-                if (stream == null) {
+                if (stream == null && !connection.streamMayHaveExisted(streamId)) {
```

Comment:
Maybe a separate issue but we use `conneciton()` and `connection` in this class...we should be consistent.